easybuild/tools/filetools.py

```diff
@@ -596,6 +597,113 @@ def download_file(filename, url, path, forced=False):
         return None
 
 
+def create_index(path, ignore_dirs=None):
+    """
+    Create index for files in specified path.
+    """
+    if ignore_dirs is None:
+        ignore_dirs = []
+
+    index = set()
+
+    if not os.path.exists(path):
+        raise EasyBuildError("Specified path does not exist: %s", path)
+    elif not os.path.isdir(path):
+        raise EasyBuildError("Specified path is not a directory: %s", path)
+
+    for (dirpath, dirnames, filenames) in os.walk(path, topdown=True):
+        for filename in filenames:
+            # use relative paths in index
+            index.add(os.path.join(dirpath[len(path) + 1:], filename))
```

Comment:
I think this gives a wrong relative path if the full path ends with '/'

Reply:
You're absolutely right, nice catch!

```
>>> path = '/Users/kehoste/work/easybuild-easyconfigs'
>>> dirpath, _, _ = list(os.walk(path))[551]
>>> dirpath[len(path) + 1:]
'easybuild/easyconfigs/r/ReFrame'
```
```
>>> path = '/Users/kehoste/work/easybuild-easyconfigs/'
>>> dirpath, _, _ = list(os.walk(path))[551]
>>> dirpath[len(path) + 1:]
'asybuild/easyconfigs/r/ReFrame'
```

Reply:
I should be using `os.path.relpath` instead:

```
>>> os.path.relpath('/Users/kehoste/work/easybuild-easyconfigs/easybuild/easyconfigs/r/ReFrame', '/Users/kehoste/work/easybuild-easyconfigs')
'easybuild/easyconfigs/r/ReFrame'
>>> os.path.relpath('/Users/kehoste/work/easybuild-easyconfigs/easybuild/easyconfigs/r/ReFrame', '/Users/kehoste/work/easybuild-easyconfigs/')
'easybuild/easyconfigs/r/ReFrame'
>>> os.path.relpath('/Users/kehoste/work/easybuild-easyconfigs/easybuild/easyconfigs/r/ReFrame', '/Users/kehoste/work/easybuild-easyconfigs////')
'easybuild/easyconfigs/r/ReFrame'
```

Reply:
fixed in b5c5b38fa